src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.cpp

```diff
@@ -17,16 +19,39 @@
 #include "Utilities/ErrorHandling/Error.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/MakeArray.hpp"
+#include "Utilities/Serialization/PupStlCpp11.hpp"
 #include "Utilities/StdHelpers.hpp"
 
 namespace domain::FunctionsOfTime {
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
 template <size_t MaxDeriv>
 QuaternionFunctionOfTime<MaxDeriv>::QuaternionFunctionOfTime(
     const double t, std::array<DataVector, 1> initial_quat_func,
     std::array<DataVector, MaxDeriv + 1> initial_angle_func,
     const double expiration_time)
     : stored_quaternions_and_times_{{t, std::move(initial_quat_func)}},
-      angle_f_of_t_(t, std::move(initial_angle_func), expiration_time) {}
+      angle_f_of_t_(t, std::move(initial_angle_func), expiration_time) {
+  stored_quaternion_size_.store(1);
+}
+
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+template <size_t MaxDeriv>
+QuaternionFunctionOfTime<MaxDeriv>::QuaternionFunctionOfTime(
+    const QuaternionFunctionOfTime<MaxDeriv>& rhs) {
+  *this = rhs;
+}
+
+template <size_t MaxDeriv>
+QuaternionFunctionOfTime<MaxDeriv>&
+// NOLINTNEXTLINE(bugprone-unhandled-self-assignment)
```

Comment:
So I did this but still got the clang-tidy error, so I left the NOLINT in. Not really sure why I still get the error because I do what [clang](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/unhandled-self-assignment.html) says to do.